### PR TITLE
drivers: spi_nrfx_spim: bring back get_nrf_spim_frequency

### PR DESCRIPTION
**Pull request by jfischer-no**
Commit 246393e83044
("drivers: spi: spi_nrfx_spim: Remove nrf_frequency_t handling")' introduced two changes, one of them is removing the function get_nrf_spim_frequency with a strange justification. This change  breaks support for peripherals written in a common way, where the maximum frequency is set to the maximum supported by the peripheral, not the controller, see shields for example.

### CODE REVIEW

**Comment by tmon-nordic** [comment]
@awojasinski-nordicsemi Please review.

File: drivers/spi/spi_nrfx_spim.c [hunk 1]
```diff
@@ -57,6 +57,39 @@ struct spi_nrfx_config {
 
 static void event_handler(const nrfx_spim_evt_t *p_event, void *p_context);
 
+static inline uint32_t get_nrf_spim_frequency(uint32_t frequency)
+{
+	/* Get the highest supported frequency not exceeding the requested one.
+	 */
+	if (frequency < KHZ(250U)) {
+		return KHZ(125U);
```
Comment (anangl) [inline drivers/spi/spi_nrfx_spim.c:65]: Could we take this opportunity and rephrase this function a bit? I think the following approach could make it clearer:
```
	if (NRF_SPIM_HAS_32_MHZ_FREQ && frequency >= MHZ(32)) {
		return MHZ(32);
	} else if (NRF_SPIM_HAS_16_MHZ_FREQ && frequency >= MHZ(16)) {
		return MHZ(16);
	} else if (frequency >= MHZ(8)) {
		return MHZ(8);

	...

	} else {
		return KHZ(125);
	}
```

**Review by awojasinski — APPROVED** [review]

**Comment by awojasinski-nordicsemi** [comment]
> @awojasinski-nordicsemi Please review.

LGTM

**Review by tmon-nordic — APPROVED** [review]

**Comment by anangl** [comment]
My proposal with refactored `get_nrf_spim_frequency`: #58001.

**Pull request closed without merging**